index.mjs

```diff
@@ -44,19 +45,26 @@ export function $(pieces, ...args) {
     }
     cmd += s + pieces[++i]
   }
-  if (verbose) {
-    printCmd(cmd)
-  }
-  let options = {
-    cwd: $.cwd,
-    shell: typeof $.shell === 'string' ? $.shell : true,
-    windowsHide: true,
-  }
-  let child = spawn($.prefix + cmd, options)
-  let promise = new ProcessPromise((resolve, reject) => {
+
+  let resolve, reject
+  let promise = new ProcessPromise((...args) => [resolve, reject] = args)
+
+  promise._run = () => {
+    if (promise.child) return
+    if (promise._prerun) promise._prerun()
+    if (verbose) {
+      printCmd(cmd)
+    }
+
+    let child = spawn(prefix + cmd, {
+      cwd,
+      shell: typeof shell === 'string' ? shell : true,
+      stdio: [promise._inheritStdin ? 'inherit' : 'pipe', 'pipe', 'pipe'],
+      windowsHide: true,
+    })
+
     child.on('exit', code => {
       child.on('close', () => {
-        if (piped) process.stdin.unpipe(child.stdin)
         let output = new ProcessOutput({
           code, stdout, stderr, combined,
```

Comment:
`combined`, `message` props are not declared`ProcessOutput` iface.

Reply:
Yes, `message` goes to the parent class. And `combined` is used inside toString() func.